relayer/pkg/chainlink/txm/txm.go

```diff
@@ -250,47 +260,54 @@ func (txm *starktxm) broadcast(ctx context.Context, publicKey *felt.Felt, accoun
+	gasConsumed := friEstimate.L1GasConsumed.BigInt(new(big.Int))
 	expandedGas := new(big.Int).Mul(gasConsumed, big.NewInt(250))
 	maxGas := new(big.Int).Div(expandedGas, big.NewInt(100))
-	tx.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGas).String())
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGas).String())
+
+	L2gasConsumed := friEstimate.L2GasConsumed.BigInt(new(big.Int))
+	L2expandedGas := new(big.Int).Mul(L2gasConsumed, big.NewInt(250))
+	L2maxGas := new(big.Int).Div(L2expandedGas, big.NewInt(100))
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L2Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(L2maxGas).String())
 
 	// pad by 150%
-	gasPrice := friEstimate.GasPrice.BigInt(new(big.Int))
+	gasPrice := friEstimate.L1GasPrice.BigInt(new(big.Int))
+	L2gasPrice := friEstimate.L2GasPrice.BigInt(new(big.Int))
 	overallFee := friEstimate.OverallFee.BigInt(new(big.Int)) // overallFee = gas_used*gas_price + data_gas_used*data_gas_price
 
 	// TODO: consider making this configurable
 	// pad estimate to 150% (add extra because estimate did not include validation)
 	gasUnits := new(big.Int).Div(overallFee, gasPrice)
 	expandedGasUnits := new(big.Int).Mul(gasUnits, big.NewInt(150))
 	maxGasUnits := new(big.Int).Div(expandedGasUnits, big.NewInt(100))
-	tx.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGasUnits).String())
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L1Gas.MaxAmount = starknetrpc.U64(starknetutils.BigIntToFelt(maxGasUnits).String())
 
 	// pad by 150%
 	expandedGasPrice := new(big.Int).Mul(gasPrice, big.NewInt(150))
 	maxGasPrice := new(big.Int).Div(expandedGasPrice, big.NewInt(100))
-	tx.ResourceBounds.L1Gas.MaxPricePerUnit = starknetrpc.U128(starknetutils.BigIntToFelt(maxGasPrice).String())
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L1Gas.MaxPricePerUnit = starknetrpc.U128(starknetutils.BigIntToFelt(maxGasPrice).String())
+
+	L2expandedGasPrice := new(big.Int).Mul(L2gasPrice, big.NewInt(150))
+	L2maxGasPrice := new(big.Int).Div(L2expandedGasPrice, big.NewInt(100))
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L2Gas.MaxPricePerUnit = starknetrpc.U128(starknetutils.BigIntToFelt(L2maxGasPrice).String())
 
 	txm.lggr.Infow("Set resource bounds", "L1MaxAmount", tx.ResourceBounds.L1Gas.MaxAmount, "L1MaxPricePerUnit", tx.ResourceBounds.L1Gas.MaxPricePerUnit)
 
-	tx.Nonce = nonce
-	// Re-sign transaction now that we've determined MaxFee
-	// TODO: SignInvokeTransaction for V3 is missing so we do it by hand
-	hash, err := account.TransactionHashInvoke(tx)
-	if err != nil {
-		return txhash, err
-	}
-	signature, err := account.Sign(ctx, hash)
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L1DataGas.MaxAmount = starknetrpc.U64(friEstimate.L1DataGasConsumed.String())
+	broadCastTxnV3.InvokeTxnV3.ResourceBounds.L1DataGas.MaxPricePerUnit = starknetrpc.U128(friEstimate.L1DataGasPrice.String())
```

Comment:
both the L1DataGas.MaxAmount and L1DataGas.MaxPricePerUnit should have the padding as well to be safe. 

Usually the MaxAmount will be 0 so multiplying it by a padding is still 0.